.github/workflows/sphinx-python-dependency-build-checks.yml

```diff
@@ -0,0 +1,43 @@
+# The purpose of this workflow file is to confirm that the Sphinx
+# virtual environment can be built from source, consequently documenting
+# the packages required in the build environment to do that.
+#
+# This is needed because some projects embeds the documentation into built
+# artifcats which involves rendering the documentation on the target
+# architecture.
+#
+# Depending on the architecture, pip may or may not have already built wheels
+# available, and as such we need to make sure building wheels from source can
+# succeed.
+name: Check and document build requirements for Sphinx venv
+on:
+  - push
+  - pull_request
+
```

Comment:
Can you add a "cancel-in-progress" (like [here](https://github.com/canonical/sphinx-docs-starter-pack/blob/a2c5c72312f6408f2d01986e70a6634d50b51fd0/.github/workflows/automatic-doc-checks.yml#L8))?

Considering that this check takes >30 min, there is no need to keep it running for every push to a PR.